creator/ingest_runs/models/validation_run.py

```diff
@@ -0,0 +1,195 @@
+import uuid
+
+import django_rq
+from django.contrib.auth import get_user_model
+from django.core.exceptions import ValidationError
+from django.db import models
+
+from creator.ingest_runs.common.model import IngestProcess
+from creator.data_reviews.models import DataReview
+from creator.files.models import Version
+
+
+User = get_user_model()
+
+
+def _get_upload_directory(validation_resultset, filename):
+    """
+    Resolves the directory where a file should be stored
+    """
+    if settings.DEFAULT_FILE_STORAGE == "django_s3_storage.storage.S3Storage":
+        prefix = f"{settings.UPLOAD_DIR}/{filename}"
+        return prefix
+    else:
+        # Validation_resultset should always have a study from its
+        # associated data_review
+        bucket = validation_resultset.study.bucket
+        directory = f"{settings.UPLOAD_DIR}/{bucket}/"
+        return os.path.join(settings.BASE_DIR, directory, filename)
+
+
+class ValidationRun(IngestProcess):
+    """
+    Request to validate file versions in a data review
+    """
+    class Meta(IngestProcess.Meta):
+        permissions = [
+            ("list_all_validationrun", "Show all validation_runs"),
+            ("cancel_validationrun", "Cancel an validation_run"),
+        ]
+
+    success = models.BooleanField(
+        null=True,
+        blank=True,
+        default=False,
+        help_text="Whether the validation run resulted in all tests passing"
+    )
+    progress = models.IntegerField(
+        null=True,
+        blank=True,
+        default=0,
+        help_text="The the percentage of the validation run that has been "
+        "completed",
+    )
+    data_review = models.ForeignKey(
+        DataReview,
+        on_delete=models.CASCADE,
+        null=True,
+        related_name="validation_runs",
+    )
+
+    @property
+    def versions(self):
+        """
+        Get file versions from the associated DataReview
+        """
+        if self.data_review:
```

Comment:
You may want to be careful with this as it may be pulling the full data review from the database. It could be expensive if there's ever a time when many validation runs get enumerated.

Reply:
Hmm I don't think so. I think that would only be true if you used `select_related` or if you actually iterate over the data review validation_runs queryset (see https://docs.djangoproject.com/en/3.1/ref/models/querysets/#when-querysets-are-evaluated)

Reply:
I'm not sure, but I'd want to test it out if it ever needs optimization. I'm suspicious of this pulling the whole row even if theoretically it only needs the foreign key column. I'd then wonder if maybe using `.exists()` would solve it or not.

Reply:
Ok let me do some testing to see

Reply:
I think we're ok here. So here are all of the queries that happen when you try to access `data_review` on a `ValidationRun`:

```
In [2]: from creator.ingest_runs.models.validation_run import ValidationRun

In [3]: vr_has_dr = ValidationRun.objects.get(data_review__isnull=False)
SELECT "ingest_runs_validationrun"."id",
       "ingest_runs_validationrun"."created_at",
       "ingest_runs_validationrun"."creator_id",
       "ingest_runs_validationrun"."input_hash",
       "ingest_runs_validationrun"."state",
       "ingest_runs_validationrun"."job_log_id",
       "ingest_runs_validationrun"."success",
       "ingest_runs_validationrun"."progress",
       "ingest_runs_validationrun"."data_review_id"
  FROM "ingest_runs_validationrun"
 WHERE "ingest_runs_validationrun"."data_review_id" IS NOT NULL

Execution time: 0.002568s [Database: default]

In [4]: vr_has_dr.data_review
SELECT "data_reviews_datareview"."kf_id",
       "data_reviews_datareview"."uuid",
       "data_reviews_datareview"."created_at",
       "data_reviews_datareview"."creator_id",
       "data_reviews_datareview"."name",
       "data_reviews_datareview"."description",
       "data_reviews_datareview"."state",
       "data_reviews_datareview"."study_id"
  FROM "data_reviews_datareview"
 WHERE "data_reviews_datareview"."kf_id" = 'DR_D2GE4DV6'

Execution time: 0.006991s [Database: default]
Out[4]: <DataReview: DataReview object (DR_D2GE4DV6)>
```

It does pull the whole row, but I don't think this matters much since the fields are just a bunch of strings essentially. The important part is it doesn't pull in any of data_review's related objects